src/nvim/api/vim.c

```diff
@@ -607,6 +607,96 @@ Array vim_get_api_info(uint64_t channel_id)
   return rv;
 }
 
+Array vim_multi_request(uint64_t channel_id, Array calls, Error *err)
+{
+  Array rv = ARRAY_DICT_INIT;
+  // make place for the number of successful requests
+  ADD(rv, INTEGER_OBJ(0));
+
+  size_t i;  // also used for freeing the variables
+  for (i = 0; i < calls.size; i++) {
+    if (calls.items[i].type != kObjectTypeArray) {
+      api_set_error(err,
+                    Validation,
+                    _("All items in calls array must be arrays"));
+      goto end;
+    }
+    Array call = calls.items[i].data.array;
+    if (call.size != 3 && call.size != 4) {
+      api_set_error(err,
+                    Validation,
+                    _("All items in calls array must be arrays of size 3 or 4"));
```

Comment:
There is a problem with that, NIL is an actual value to which the return value could be compared. If we insist of a fixed size we either need a bool flag `[name, args, sendret, assertret, assertval]` or maybe merge with `sendret` (because if you assert the value, you don't want it returned, unless the assertion fails).